core/src/test/java/org/apache/iceberg/TestSchemaUnionByFieldName.java

```diff
@@ -75,7 +75,7 @@ private static NestedField[] primitiveFields(
                 optional(
                     atomicInteger.incrementAndGet(),
                     type.toString(),
-                    Types.fromPrimitiveString(type.toString())))
+                    Types.fromTypeString(type.toString())))
```

Comment:
It looks like this is correct, but do you want to add `VariantType.get()` to the list of types that are tested as well?